firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayPanelViewController.swift

```diff
@@ -255,6 +261,23 @@ final class TabDisplayPanelViewController: UIViewController,
         }
     }
 
+    private func updateInsets() {
+        if isCompactLayout {
+            let emptyView = emptyPrivateTabsView as? InsetUpdatable
+            let bottomInset = if (emptyView as? ExperimentEmptyPrivateTabsView) != nil {
+                DefaultTabTrayUtils().segmentedControlHeight + view.safeAreaInsets.bottom
+            } else {
+                DefaultTabTrayUtils().segmentedControlHeight
+            }
+
+            emptyView?.updateInsets(top: 0, bottom: bottomInset)
+            tabDisplayView.updateInsets(top: 0, bottom: DefaultTabTrayUtils().segmentedControlHeight)
+        } else {
+            (emptyPrivateTabsView as? InsetUpdatable)?.updateInsets(top: view.safeAreaInsets.top, bottom: 0)
```

Comment:
Same same 👀 `EmptyPrivateTabView` can be `InsetUpdatable` to avoid this cast I think